src/compiler/emitter.ts

```diff
@@ -1346,6 +1347,8 @@ namespace ts {
             else {
                 writeToken(SyntaxKind.OpenBraceToken, node.pos, /*contextNode*/ node);
                 emitBlockStatements(node);
+                // We have to call emitLeadingComments explicitly here because otherwise leading comments of the close brace token will not be emitted
+                emitLeadingComments(node.statements.end, /*isEmittedNode*/true);

```

Comment:
do you mean make a call like this `emitLeadingCommentsOfPosition(nodeList.pos)`?